docs/presigned_urls.md

```diff
@@ -0,0 +1,354 @@
+
+## Prerequisites
+
+### AWS Security Credentials Setup
+
+In order to sign programmatic requests to an AWS account through the AWS Command Line Interface (CLI) or AWS APIs, the convention is to create a `.aws` directory in the computer's home folder, containing `config` and `credentials` files. The location of this directory is described in further detail in the AWS documentation [here](https://docs.aws.amazon.com/sdkref/latest/guide/file-location.html). 
+
+These shared files should contain AWS access keys required for creation of presigned URLs (and overall AWS account access); these keys consist of an `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`. The Access Key ID should not be shared casually, but the Secret Access Key should never be shared and is considered highly confidential. For security reasons, these keys should be [rotated regularly](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html#Using_RotateAccessKey).
+
+Only [users with appropriate security credentials](https://docs.aws.amazon.com/AmazonS3/latest/userguide/using-presigned-url.html#who-presigned-url) can create presigned URLs: [Identity and Access Management (IAM) users](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users.html), [IAM instance profiles](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2_instance-profiles.html) (an IAM role for EC2 instances), and credentials generated via the AWS Security Token Service. The difference between these credentials presigned URLs generated with credentials for an (1) IAM instance profile is valid for up to 6 hours, (2) AWS Security Token Service is valid up to 36 hours when signed with permanent credentials, and (3) IAM user is valid up to 7 days. 
+
+For whichever set of credentials chosen, generate the appropriate AWS access ID-key pair. For the purposes of this documentation, generation of presigned URLs for an IAM user will be described, but follow AWS documentation to generate these keys for [EC2 instances](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-key-pairs.html) and through the [AWS Security Token Service](https://docs.aws.amazon.com/STS/latest/APIReference/welcome.html). For IAM user profiles, a user-friendly way to generate access keys is using the [AWS console](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html#Using_CreateAccessKey), by navigating to the **Security credentials** tab under the appropriate user within the **IAM** (Identity and Access Management) area in the console and generating secret access keys. Save these values in the `~/.aws/credentials` file (`C:\Users\%USERPROFILE%\.aws\credentials` for Windows users). It will look similar to the following:
```

Comment:
for the windows part. Has anyone from our team tried to set it up on Windows? If not, maybe it's too detailed

Reply:
I don't think anyone on our team has, since we're all using Macs to my knowledge. I'll remove this detail and let users of other OSes refer to the AWS docs